### PR TITLE
Fix/ctaa google zero risk glitch part 2

### PR DESCRIPTION
**Pull request by uliluckas**
## Changes

Make sure worker is not blocked by a previous, potentially failed, worker of same unique name.

Also fix order of prerequisite checks to check for 2nd call-back first

### CODE REVIEW

**Review by falkorichter — DISMISSED** [review]
Looks great

File: app/src/main/java/at/roteskreuz/stopcorona/model/receivers/ExposureNotificationBroadcastReceiver.kt [hunk 1]
```diff
@@ -33,9 +33,32 @@ import timber.log.Timber
 /**
  * Broadcast receiver for callbacks from exposure notification API.
  */
-class ExposureNotificationBroadcastReceiver : BroadcastReceiver(), KoinComponent {
+class ExposureNotificationBroadcastReceiver : BroadcastReceiver() {
 
-    private val workManager: WorkManager by inject()
+    companion object : KoinComponent {
```
Comment (dusanjencik) [inline app/src/main/java/at/roteskreuz/stopcorona/model/receivers/ExposureNotificationBroadcastReceiver.kt]: Static injecting brokes whole DI flow. If you need workManager inside, provide it as a parameter and inject it in the caller location.

File: app/src/main/java/at/roteskreuz/stopcorona/model/receivers/ExposureNotificationBroadcastReceiver.kt [hunk 2]
```diff
@@ -33,9 +33,32 @@ import timber.log.Timber
...
+        private val workManager: WorkManager by inject()
+
+        fun onExposureStateUpdated(context: Context, token: String) {
```
Comment (dusanjencik) [inline app/src/main/java/at/roteskreuz/stopcorona/model/receivers/ExposureNotificationBroadcastReceiver.kt]: Moving implementation to the static block brokes Interface segregation principle. This is not a good practice. Please move it back to the `onReceive` method.

File: app/src/main/java/at/roteskreuz/stopcorona/model/workers/DelayedExposureBroadcastReceiverCallWorker.kt [hunk 3]
```diff
@@ -43,23 +42,21 @@ class DelayedExposureBroadcastReceiverCallWorker(
                 .build()
 
             workManager.enqueueUniqueWork(
-                TAG,
-                ExistingWorkPolicy.APPEND,
+                "$TAG-$token",
```
Comment (dusanjencik) [inline app/src/main/java/at/roteskreuz/stopcorona/model/workers/DelayedExposureBroadcastReceiverCallWorker.kt:47]: If you have a unique TAG, we cannot cancel it easily if we would like to.

File: app/src/main/java/at/roteskreuz/stopcorona/model/workers/DelayedExposureBroadcastReceiverCallWorker.kt [hunk 4]
```diff
@@ -43,23 +42,21 @@ class DelayedExposureBroadcastReceiverCallWorker(
...
         val token = inputData.getString(ARGUMENT_TOKEN)
         Timber.d("ENStatusUpdates: Timeout for $token expired, let's check if it has been processed")
         // fake call to be sure that the zero risk is processed
         token?.let {
-            receiver.onExposureStateUpdated(applicationContext, token)
+            ExposureNotificationBroadcastReceiver.onExposureStateUpdated(applicationContext, token)
         }
 
-        return Result.failure()
+        return Result.success()
```
Comment (dusanjencik) [inline app/src/main/java/at/roteskreuz/stopcorona/model/workers/DelayedExposureBroadcastReceiverCallWorker.kt:65]: Nice catch 👍

**Review by dusanjencik — CHANGES_REQUESTED** [review]

File: app/src/main/java/at/roteskreuz/stopcorona/model/receivers/ExposureNotificationBroadcastReceiver.kt (hunk 2 above)
Comment (uliluckas) [reply app/src/main/java/at/roteskreuz/stopcorona/model/receivers/ExposureNotificationBroadcastReceiver.kt]: ok

File: app/src/main/java/at/roteskreuz/stopcorona/model/receivers/ExposureNotificationBroadcastReceiver.kt (hunk 1 above)
Comment (uliluckas) [reply app/src/main/java/at/roteskreuz/stopcorona/model/receivers/ExposureNotificationBroadcastReceiver.kt]: ok

**Review by dusanjencik — APPROVED** [review]
Nice 👍

**Pull request merged**